Document breaking change rule between empty and non-empty struct

I hit this for real. https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md does not mention this source-breaking change:

```cs
using System;

public struct S { }

public class C
{
    public void M()
    {
        S s;
        Console.WriteLine(s);
    }
}
```

###### [(sharplab)](https://sharplab.io/#v2:EYLgHgbALANALiAbgUwE4EsBmBPAPgVwGd0A7AcwAIBlbQuZAWwG4BYAKHYAd9gAbdAMYU6qfALjUKAbwoBfdlx78hA3gENChCgGF2U9hUMVufQRUQB7dABMKAWQAUASgNH9bI58mFWHr4e0LEkILXmQAOgB1DHoAGVJkB0InX095NnSgA==)

After adding a field:

```cs
using System;

public struct S
{
    private readonly int foo;
}

public class C
{
    public void M()
    {
        S s;
        Console.WriteLine(s); // error CS0165: Use of unassigned local variable 's'
    }
}
```

###### [(sharplab)](https://sharplab.io/#v2:EYLgHgbALANALiAbgUwE4EsBmBPAPgVwGd0A7AcwAIBlbQuZAWwG4BYAKHYAd9gAbdAMYU6qfALjV2Ab3YU5FThkQBDehVTJlAEwD2JXtgqkJmHTtZsAvuy49+Qgb2WFCFAMLTZ87n0EVEOuhaFACyABQAlF5yMmzy8dTCFgnybnqEOrzIAHQA6hj0ADKkyGGEEcny1lbsQA)

--- a/Documentation/coding-guidelines/breaking-change-rules.md
+++ b/Documentation/coding-guidelines/breaking-change-rules.md
@@ -211,6 +211,10 @@ successfully bind to that overload, if simply passing an `int` value. However, i
 
 * Adding or removing `static` keyword from a member
 
+* Adding a field to a struct that previously had no state
+
+    Definite assignment rules allow use of uninitialized variables so long as the variable type is a stateless struct. If the struct is made stateful, code could now end up with uninitialized data. This is both potentially a source breaking and binary breaking change.
+
 ### Signatures
 &#10003; **Allowed**
 * Adding `params` to a parameter

